plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileSystemCache.java

```diff
@@ -130,6 +130,9 @@ private static FileSystem createFileSystem(URI uri, Configuration conf)
         }
         FileSystem original = (FileSystem) ReflectionUtils.newInstance(clazz, conf);
         original.initialize(uri, conf);
+        if (original.getClass().getSimpleName().equals("HoodieWrapperFileSystem")) {
+            return original;
+        }
```

Comment:
Hudi does extra wrapping on FileSystem. The caching layer does not work with it. So we dont to have the filesystem objects cached. What is a better way to accomplish this target?

Reply:
I generally agree that there should not be table formats (Hudi/Iceberg/Delta) specific code in trino-hive. But, it's hard to completely decouple. We already have [annotation-based](https://github.com/trinodb/trino/blob/4338264e170e0a63770460ca6acd23a18cff4988/plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java#L771) check to support querying hudi tables through the hive connector. I also see some checks related to Iceberg/Delta table in `HiveUtil`.

In this particular case, we need to make the caching work with `HoodieWrapperFileSystem`. Since the above conditional only affects hudi connector and does not change the hive connector critical path, can we take this task as a follow-up?

Reply:
We created a seperate PR to solve this issue, and make the current PR clean. https://github.com/trinodb/trino/pull/12700 Could you take a review on it? @losipiuk cc @codope

Reply:
Done in preparatory commits of this PR.